test/http_responses_exceptions_test.dart

```diff
@@ -8,60 +8,76 @@ import 'package:cronet/cronet.dart';
 import 'package:test/test.dart';
 
 void main() {
-  late HttpClient client;
-  late HttpServer server;
-  setUp(() async {
-    client = HttpClient();
-    server = await HttpServer.bind(InternetAddress.anyIPv6, 5255);
-    server.listen((HttpRequest request) {
-      final paths = request.uri.pathSegments;
-      request.response.statusCode = int.parse(paths[0]);
-      request.response.close();
+  group('response_HttpException_test', () {
+    late HttpClient client;
+    late HttpServer server;
+    setUp(() async {
+      client = HttpClient();
+      server = await HttpServer.bind(InternetAddress.anyIPv6, 5255);
```

Comment:
Do all the tests use this port? Maybe you should run the tests like `dart run test --concurrency=1` (and also mention this in readme) otherwise this port could be in use since package:test runs tests concurrently (4 threads by default).

Reply:
If you use `0` as the port you probably don't have to worry about concurrency anymore.